classes/form/edit_conditions.php

```diff
@@ -153,6 +162,19 @@ public function definition() {
             )
         );
 
+        $repeatarray[] = $mform->createElement(
```

Comment:
Could we swap this to the user search w/ Ajax selector? will be much easier to use

Reply:
Good point, it looks much better now.

Reply:
Hang on, I'm getting `Exception - Serialization of 'Closure' is not allowed` in some scenario. Will have a look what's happening there

Reply:
Sadly this was revealed when everything else is implemented :(
It seems like moodle form doesn't like cloning elements with functions https://github.com/moodle/moodle/blob/master/lib/moodlelib.php#L9430

```
Stack trace:
line 9430 of /lib/moodlelib.php: Exception thrown
line 9430 of /lib/moodlelib.php: call to serialize()
line 1178 of /lib/formslib.php: call to fullclone()
line 213 of /admin/tool/abconfig/classes/form/edit_conditions.php: call to moodleform->repeat_elements()
line 214 of /lib/formslib.php: call to tool_abconfig\form\edit_conditions->definition()
line 50 of /admin/tool/abconfig/edit_conditions.php: call to moodleform->__construct()
```

Reply:
Ok, this seems to be sorted by moving the callable function into a separate static method.
Happy to have this reviewed @Peterburnett